src/audio.rs

```diff
@@ -0,0 +1,76 @@
+use cpal::traits::{DeviceTrait, HostTrait};
+use cpal::{Device, InputCallbackInfo, SampleFormat, StreamConfig, SupportedStreamConfig};
+use std::sync::{self, mpsc::Receiver, mpsc::Sender};
+
+pub fn start_audio_loop() -> (cpal::Stream, Receiver<i16>) {
+    let audio_device = get_audio_device();
+    let input_config = get_input_config(&audio_device);
+    let sample_format = input_config.sample_format();
+    let config = input_config.into();
+
+    println!("Sample format: {:?}", sample_format);
+
+    start_stream(&config, &audio_device, &sample_format)
+}
+
+fn get_audio_device() -> Device {
+    let host = cpal::default_host();
+    let mut devices = host.devices().expect("Host has no audio device");
+    devices
+        .find(|device| device.name().expect("Audio device has no name") == "pulse")
```

Comment:
Why care that devices would have no names if we're searching for only one device with a valid name?

```suggestion
        .find(|device| device.name() == Ok("pulse"))
```

Reply:
@DIGarnier En fait c'est juste le message d'erreur que j'ai mis qui est missleading, si le expect fonctionne pas c'est que y'a eu un erreur quand on a ask au OS de nous donner le nom, ca devrait jamais arriver.
J'ai expliqué a thom ici https://github.com/TurboAudio/turbo_audio/pull/1#discussion_r985270499